tests/test_config.py

```
@@ -0,0 +1,345 @@
+class TestCrawlerrSettings:
+    """Test the main configuration class."""
+
+    @pytest.mark.unit
+    def test_environment_loading(self):
+        """Test that configuration correctly loads from environment (.env file)."""
+        config = CrawlerrSettings()
+
+        # Test that values are loaded from .env file (not hardcoded defaults)
+        assert config.server_host == "0.0.0.0"  # From .env SERVER_HOST
+        assert config.server_port == 8010  # From .env SERVER_PORT
+        assert config.debug == True  # From .env DEBUG=true
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `config.debug:` for truth checks

Replace with `config.debug`

(E712)